File: LeetCode/14.longest-common-prefix.cpp
/*
 * @lc app=leetcode id=14 lang=cpp
 *
 * [14] Longest Common Prefix
 */
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;
// @lc code=start
class Solution
{
public:
    
string longestCommonPrefix(vector<string> &strs) {
    if (strs.empty()) {
        return "";
    }

    string prefix = strs[0];

    for (int i = 1; i < strs.size(); i++) {
        int j = 0;
        while (j < prefix.length() && j < strs[i].length() && prefix[j] == strs[i][j]) {
            j++;
        }

        prefix = prefix.substr(0, j);

        if (prefix.empty()) {
            break;  // No common prefix found
        }
    }

    return prefix;
}

    
};
// @lc code=end

File: LeetCode/14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python3
#
# [14] Longest Common Prefix
#

from typing import List

# @lc code=start
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs or len(min(strs)) == 0:
            return ""

        n = ""


        for i in range(len(min(strs))):
            curr = ''
            isAllTheSame = True
            for c in range(len(strs)):
                if c == 0:
                    curr = strs[c][i]
                else:
                    if curr != strs[c][i]:
                        return n
            n += curr
        return n
        
                        
s = Solution()
print(s.longestCommonPrefix(["a"]))
# @lc code=end

